File: api/src/main/java/org/axostudio/axohologram/api/hologram/LineType.java
package org.axostudio.axohologram.api.hologram;

import java.util.Locale;

public enum LineType {
    TEXT,
    ITEM,
    BLOCK,
    COMPOSITE,
    ANIMATED_TEXT,
    UNKNOWN;

    public static LineType fromString(String raw) {
        if (raw == null || raw.trim().isEmpty()) {
            return TEXT;
        }
        try {
            return valueOf(raw.trim().toUpperCase(Locale.ROOT));
        } catch (IllegalArgumentException e) {
            return UNKNOWN;
        }
    }
}

File: src/main/kotlin/org/axostudio/axohologram/core/hologram/line/LineType.kt
package org.axostudio.axohologram.core.hologram.line

import java.util.Locale

enum class LineType {
    TEXT,
    ITEM,
    BLOCK,
    COMPOSITE,
    ANIMATED_TEXT,
    UNKNOWN;

    companion object {
        @JvmStatic
        fun fromString(raw: String?): LineType {
            if (raw.isNullOrBlank()) return TEXT
            return runCatching {
                valueOf(raw.trim().uppercase(Locale.ROOT))
            }.getOrDefault(UNKNOWN)
        }
    }
}
